packages/lesswrong/components/posts/PostsHighlight.tsx

```diff
@@ -53,10 +71,80 @@ const PostsHighlight = ({post, maxLengthWords, forceSeeMore=false, classes}: {
       description={`post ${post._id}`}
       nofollow={(post.user?.karma || 0) < nofollowKarmaThreshold.get()}
     />
-    {loading && expanded && <Components.Loading/>}
+    {expandedLoading && expanded && <Components.Loading/>}
   </Components.ContentStyles>
+}
+
+const ForeignPostsHighlight = ({post, maxLengthWords, forceSeeMore=false, classes}: {
+  post: PostsList & PostWithForeignId,
+  maxLengthWords: number,
+  forceSeeMore?: boolean,
+  classes: ClassesType,
+}) => {
+  const {loading, error, combinedPost} = useForeignCrosspost(post, foreignFetchProps);
+  if (error) {
+    throw error;
```

Comment:
Does this get caught more locally than "the top-level app layer"?  I think in most places we get an error from a graphql query we don't throw but render some sort of minimal error component instead.